tests/unit/python/foglamp/tasks/purge/test_purge.py

```diff
@@ -0,0 +1,256 @@
+                              ({"retainUnsent": {"value": "True"}, "age": {"value": "0"}, "size": {"value": "bla"}},
+                               "size")])
+    def test_purge_data_invalid_conf(self, event_loop, conf, expected_error_key):
+        """Test that purge_data raises exception when called with invalid configuration"""
+
+        @asyncio.coroutine
+        def mock_audit_info():
+            return ""
+
+        mockStorageClient = MagicMock(spec=StorageClient)
+        mockAuditLogger = AuditLogger(mockStorageClient)
+
+        with patch.object(FoglampProcess, '__init__'):
+            with patch.object(mockAuditLogger, "__init__", return_value=None):
+                p = Purge(loop=event_loop)
+                p._logger = logger
+                p._logger.info = MagicMock()
+                p._logger.error = MagicMock()
+                p._storage = MagicMock(spec=StorageClient)
+                p._readings_storage = MagicMock(spec=ReadingsStorageClient)
+                audit = p._audit
+                with patch.object(p._readings_storage, 'purge', side_effect=self.store_purge):
+                    with patch.object(audit, 'information', return_value=mock_audit_info()):
+                        # Test the code block when purge failed because of invalid configuration
+                        with pytest.raises(ValueError):
+                            p.purge_data(conf)
+                        p._logger.error.assert_called_with('Configuration item {} bla should be integer!'.
+                                                           format(expected_error_key))
+
+    def test_run(self, event_loop):
+        """Test that run calls all units of purge process"""
+        @asyncio.coroutine
+        def mock_audit_info():
+            return ""
+
+        mockStorageClient = MagicMock(spec=StorageClient)
+        mockAuditLogger = AuditLogger(mockStorageClient)
+
+        with patch.object(FoglampProcess, '__init__'):
+            with patch.object(mockAuditLogger, "__init__", return_value=None):
+                p = Purge(loop=event_loop)
+                config = "Some config"
+                p._logger.exception = MagicMock()
+                with patch.object(p, 'set_configuration', return_value=config) as mock_set_config:
+                    with patch.object(p, 'purge_data', return_value=(1, 2)) as mock_purge_data:
+                        with patch.object(p, 'write_statistics') as mock_write_stats:
+                            p.run()
+                # Test the positive case when no error in try block
+                mock_set_config.assert_called_once_with()
+                mock_purge_data.assert_called_once_with(config)
+                mock_write_stats.assert_called_once_with(1, 2)
+
+    def test_run_exception(self, event_loop):
+        """Test that run calls all units of purge process and checks the exception handling"""
+        @asyncio.coroutine
+        def mock_audit_info():
+            return ""
```

Comment:
coroutine is unused?

Reply:
returns "" but not unused, patch uses this async coroutine